src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -202,6 +211,33 @@ public class RedundantModifierCheck
         TokenTypes.ABSTRACT,
     };
 
+    /**
+     *  Constant for jdk 21 version number.
+     */
+    private static final int JDK_21 = 21;
+
+    /**
+     * Set the JDK version that you are using.
+     *
+     */
+    private String jdkVersion = String.valueOf(JDK_21);
+
+    /**
+     * Java language level based on the JDK version.
+     */
+    private JavaLanguageLevel javaLanguageLevel = JavaLanguageLevel.getLevel(jdkVersion);
+
+    /**
+     * Setter to set the JDK version that you are using.
+     *
+     * @param jdkVersion the Java version
+     * @since 10.18.0
+     */
+    public void setJdkVersion(String jdkVersion) {
+        this.jdkVersion = jdkVersion;
```

Comment:
Why do we need to save the string? It looks like we fully rely on the enumeration.

Reply:
no reason just dummy saving. shouldn't we save it? and why?

Reply:
If we are switching to the int, then I feel we don't need this. To me this looks like it exists only to make default value on documentation easier to print.

Setter should be able to be a String while a single field is an int. They are not required to be the same type.